patches/net/minecraft/core/MappedRegistry.java.patch

```diff
@@ -41,14 +41,37 @@
        }
  
        this.byKey.put(p_256594_, reference);
-@@ -162,6 +_,7 @@
+       this.byLocation.put(p_256594_.location(), reference);
+       this.byValue.put(p_256374_, reference);
+-      this.byId.size(Math.max(this.byId.size(), p_256563_ + 1));
++      resizeListFast(this.byId, Math.max(this.byId.size(), p_256563_ + 1));
+       this.byId.set(p_256563_, reference);
+       this.toId.put(p_256374_, p_256563_);
+       if (this.nextId <= p_256563_) {
+@@ -162,9 +_,23 @@
        this.lifecycles.put(p_256374_, p_256469_);
        this.registryLifecycle = this.registryLifecycle.add(p_256469_);
        this.holdersInOrder = null;
 +      this.addCallbacks.forEach(addCallback -> addCallback.onAdd(this, p_256563_, p_256594_, p_256374_));
        return reference;
     }
  
++   private static <T> void resizeListFast(ObjectList<T> list, int size) {
++      if (list.size() < size) {
++         // write null entries to fill size, to match the behavior of ObjectList.size(int)
++         // ObjectList.add will grow the list intelligently rather than copying the array each time
++         while (list.size() < size) {
++            list.add(null);
++         }
++      } else if(list.size() > size) {
```

Comment:
If we switch the order of these if branches so that this one is first, then the other one can just be a simple `else` and have the other check just be done when the while statement is first evaluated.